File: 0547-number-of-provinces/0547-number-of-provinces.java
class Solution {
    public int findCircleNum(int[][] isConnected) {
        int n = isConnected.length;
        int cnt = 0;
        boolean[] vis = new boolean[n];
        for(int i=0; i<n; i++){
            if(!vis[i]){
                cnt++;
                dfs(i, isConnected, vis);
            }
        }
        return cnt;
    }
    void dfs(int node, int[][] isConnected, boolean[] vis){
        vis[node] = true;
        for(int i=0; i<isConnected.length; i++){
            if(isConnected[node][i]==1 && !vis[i]) dfs(i, isConnected, vis);
        }
    }
}

File: 0547-number-of-provinces/0547-number-of-provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        n = len(isConnected)
        vis = [False] * n
        cnt = 0
        for i in range(n):
            if not vis[i]:
                cnt += 1
                self.dfs(i, isConnected, vis)
        return cnt
    def dfs(self, node, isConnected, vis):
        vis[node] = True
        for i in range(len(isConnected)):
            if not vis[i] and isConnected[node][i]: self.dfs(i, isConnected, vis)